README.md

```diff
@@ -95,26 +95,37 @@ To make calls, you must create a project and segment in the IronCore Admin Conso
 
 ### Prerequisites
 
-- [Rust toolchain](https://www.rust-lang.org/tools/install) installed.
-- [cross](https://github.com/rust-embedded/cross) installed.
-- Android SDK 29. You can get the command line SDK [here](https://developer.android.com/studio) (scroll down to "Command line tools only") and then use `sdkmanager` (found in tools/bin) to install additional prerequisites:
+- Install [Rust toolchain](https://www.rust-lang.org/tools/install).
+- Install [cross](https://github.com/rust-embedded/cross).
+- Edit `$HOME/.gradle/gradle.properties` to include `sdk.dir=PATH_TO_ANDROID_SDK`.
+- Install Android SDK 29. You can get the command line SDK [here](https://developer.android.com/studio) (scroll down to "Command line tools only"). The extracted `tools` folder must follow a specific folder hierarchy. We recommend `AndroidCLI/cmdline-tools/tools`. You are then able to use `sdkmanager` (found in `tools/bin`) to install additional prerequisites:
 
   ```bash
-  ./sdkmanager --sdk_root=PATH_TO_SDK_INSTALL_LOCATION --install "build-tools;29.0.3" platform-tools "platforms;android-29"
+  ./sdkmanager "build-tools;29.0.3" "platform-tools" "platforms;android-29"
   ```
 
-- Edit `$HOME/.gradle/gradle.properties` to point the `sdk.dir` to the location of the Android SDK.
-
 ### Building
 
-From `ironoxide-java/android`, run `build.sh`. The output AAR file will be in `android/ironoxide-android/build/outputs/aar`.
+From the repository root, run `android/build.sh`. The output AAR file will be in `android/ironoxide-android/build/outputs/aar`.
+
+## Testing
+
+### Prerequisites
+
+- Successfully run `build.sh` by following the steps in [Build from Source](#build-from-source-1).
+  - This will create the java files and the `.so` files required for the `x86`, `x86_64`, and `arm64-v8a` architectures. If testing on a different architecture, you can find the Rust target to compile to [here](https://forge.rust-lang.org/release/platform-support.html).
+- An Android emulator running, or a compatible Android phone connected.
+  - To start an emulator using the command line tools, follow these steps from the folder `AndroidCLI/cmdline-tools/tools/bin`:
+    1. `./sdkmanager "emulator" "system-images;android-29;google_apis;x86_64"`
+    2. `./avdmanager create avd -n pixel_3 -k "system-images;android-29;google_apis;x86_64" -d pixel_3`
+    3. `../../../emulator/emulator -avd pixel_3 -no-window -gpu swiftshader_indirect -no-snapshot -noaudio -no-boot-anim`
```

Comment:
It might be nice to add something to this line saying that it might display some text/warnings, but the emulator will be up and running and waiting for you to run the tests in the next step. Given how long some of the above operations take, it'd be good to explain that this command won't complete so people aren't waiting here.

Reply:
Good call. I added a point about this.